src/gsi/mod_fv3_lola.f90

```diff
@@ -164,431 +168,641 @@ subroutine generate_anl_grid(nx,ny,grid_lon,grid_lont,grid_lat,grid_latt)
+      allocate(region_lat(nlat,nlon),region_lon(nlat,nlon))
+      allocate(fv3dx(nxa,nya),fv3dx1(nxa,nya),fv3dy(nxa,nya),fv3dy1(nxa,nya) )
+      allocate(fv3ix(nxa,nya),fv3ixp(nxa,nya),fv3jy(nxa,nya),fv3jyp(nxa,nya) )
+      allocate( a3dx( ny, nx),a3dx1(ny,nx),a3dy(ny,nx),a3dy1(ny,nx) )
+      allocate( a3ix( ny, nx),a3ixp(ny,nx),a3jy(ny,nx),a3jyp(ny,nx) )
+      allocate(cangu( nx,ny+1),sangu(nx,ny+1),cangv(nx+1,ny),sangv(nx+1,ny) )
+
+      rewind(inunit)
+      !time_beg=MPI_Wtime()
+      read(inunit) nlat,nlon,nxa,nya,  &
+                   region_dx,region_dy,region_dxi,region_dyi,  &
+                   coeffx,coeffy,region_lat,region_lon,  &
+                   fv3dx,fv3dx1,fv3dy,fv3dy1,fv3ix,fv3ixp,fv3jy,fv3jyp, &
+                   a3dx,a3dx1,a3dy,a3dy1,a3ix,a3ixp,a3jy,a3jyp, &
+                   cangu,sangu,cangv,sangv
+      !time_end=MPI_Wtime()
+      !call MPI_Reduce(time_end-time_beg, walltime, 1, MPI_REAL8, MPI_MAX, 0, nodeComm, ierr)
+      !if(mype==0) write(6,'("Maximum Walltime to read anl_grid " f15.4)') walltime
+
+      close(inunit)
+    else
 
 !   create xc,yc,zc for the cell centers.
-  allocate(xc(nx,ny))
-  allocate(yc(nx,ny))
-  allocate(zc(nx,ny))
-  allocate(gclat(nx,ny))
-  allocate(gclon(nx,ny))
-  allocate(gcrlat(nx,ny))
-  allocate(gcrlon(nx,ny))
-  do j=1,ny
-     do i=1,nx
-        xc(i,j)=cos(grid_latt(i,j)*deg2rad)*cos(grid_lont(i,j)*deg2rad)
-        yc(i,j)=cos(grid_latt(i,j)*deg2rad)*sin(grid_lont(i,j)*deg2rad)
-        zc(i,j)=sin(grid_latt(i,j)*deg2rad)
-     enddo
-  enddo
+      allocate(xc(nx,ny))
+      allocate(yc(nx,ny))
+      allocate(zc(nx,ny))
+      allocate(gclat(nx,ny))
+      allocate(gclon(nx,ny))
+      allocate(gcrlat(nx,ny))
+      allocate(gcrlon(nx,ny))
+      !$omp parallel do default(none),private(j,i) &
+      !$omp shared(ny,nx,xc,grid_latt,grid_lont,deg2rad,yc,zc)
+      do j=1,ny
+       do i=1,nx
+          xc(i,j)=cos(grid_latt(i,j)*deg2rad)*cos(grid_lont(i,j)*deg2rad)
+          yc(i,j)=cos(grid_latt(i,j)*deg2rad)*sin(grid_lont(i,j)*deg2rad)
+          zc(i,j)=sin(grid_latt(i,j)*deg2rad)
+       enddo
+      enddo
+      !$omp end parallel do
 
 !  compute center as average x,y,z coordinates of corners of domain --
 
-  xcent=quarter*(xc(1,1)+xc(1,ny)+xc(nx,1)+xc(nx,ny))
-  ycent=quarter*(yc(1,1)+yc(1,ny)+yc(nx,1)+yc(nx,ny))
-  zcent=quarter*(zc(1,1)+zc(1,ny)+zc(nx,1)+zc(nx,ny))
+      xcent=quarter*(xc(1,1)+xc(1,ny)+xc(nx,1)+xc(nx,ny))
+      ycent=quarter*(yc(1,1)+yc(1,ny)+yc(nx,1)+yc(nx,ny))
+      zcent=quarter*(zc(1,1)+zc(1,ny)+zc(nx,1)+zc(nx,ny))
 
-  rnorm=one/sqrt(xcent**2+ycent**2+zcent**2)
-  xcent=rnorm*xcent
-  ycent=rnorm*ycent
-  zcent=rnorm*zcent
-  centlat=asin(zcent)*rad2deg
-  centlon=atan2(ycent,xcent)*rad2deg
+      rnorm=one/sqrt(xcent**2+ycent**2+zcent**2)
+      xcent=rnorm*xcent
+      ycent=rnorm*ycent
+      zcent=rnorm*zcent
+      centlat=asin(zcent)*rad2deg
+      centlon=atan2(ycent,xcent)*rad2deg
```

Comment:
Looks like some additional indents were added in `mod_fv3_lola.f90`. Not sure if that is intended or not.